### PR TITLE
[437497] Prevent NPE in GlobalURIEditorOpener 

### PR DESCRIPTION
**Pull request by cdietrich**
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=437497

File: plugins/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/GlobalURIEditorOpener.java [hunk 1]
```diff
@@ -73,7 +76,10 @@ public IEditorPart open(URI uri, boolean select) {
...
 		IResourceServiceProvider resourceServiceProvider = resourceServiceProviderRegistry.getResourceServiceProvider(referenceOwnerURI.trimFragment());
 		if(resourceServiceProvider instanceof IResourceUIServiceProvider) {
-			return ((IResourceUIServiceProvider) resourceServiceProvider).getURIEditorOpener().open(referenceOwnerURI, reference, indexInList, select);
+			IURIEditorOpener editorOpener = ((IResourceUIServiceProvider) resourceServiceProvider).getURIEditorOpener();
+			if (editorOpener != null) {
+				return editorOpener.open(referenceOwnerURI, reference, indexInList, select);
+			}
 		}
 		return openDefaultEditor(referenceOwnerURI, reference, indexInList, select);
 	}
```
Comment (LorenzoBettini) [inline plugins/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/GlobalURIEditorOpener.java]: The two modifications highlight that the lines are basically the same; would it be worth taking the opportunity to extract the logic in a method?

Comment (cdietrich) [inline plugins/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/GlobalURIEditorOpener.java]: the method called is different

Comment (LorenzoBettini) [inline plugins/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/GlobalURIEditorOpener.java]: Sorry, I was referring to the logic for retrieving the editorOpener (starting from the instanceof)

**Pull request merged**